src/nvim/os/os_defs.h

```diff
@@ -18,11 +18,15 @@
 #endif
 
 #if defined(UFS_MAXNAMLEN) && !defined(MAXNAMLEN)
-# define MAXNAMLEN UFS_MAXNAMLEN    /* for dynix/ptx */
+# define MAXNAMLEN UFS_MAXNAMLEN    // for dynix/ptx
```

Comment:
At least one reason is that it can silently break multi-line macros: http://stackoverflow.com/a/1511383/152142

TBH I don't remember if there is any technical reason to disallow `//` in single-line macros like this.